pkg/thing/interactors/unregister_thing.go

```diff
@@ -0,0 +1,41 @@
+package interactors
+
+import (
+	"errors"
+)
+
+// Unregister runs the use case to remove a registered thing
+func (i *ThingInteractor) Unregister(authorization, id string) error {
+	i.logger.Debug("Executing unregister thing use case")
+
+	if authorization == "" {
+		return errors.New("authorization key not provided")
+	}
+
+	if id == "" {
+		return errors.New("thing's id not provided")
+	}
+
+	err := i.thingProxy.RemoveThing(authorization, id)
+	if err != nil {
+		msg := err.Error()
+		sendErr := i.clientPublisher.SendUnregisteredDevice(id, &msg)
+		if sendErr != nil {
+			return sendErr
+		}
+		return err
```

Comment:
@lcbm I didn't add a logger before any of the error returns. This is because the the method whose calls this interactor (in `msg_hander.go`) logs every returned error, so I thought that log the error in interactor itself would be ambiguous.

Reply:
Makes sense! :smile: 
Thanks for sharing :wink:

Reply:
I usually prefer to log only when necessary, but this is just my personal preference. Both approaches are correct.
Dave Cheney had written about this subject [here](https://dave.cheney.net/practical-go/presentations/qcon-china.html#_only_handle_an_error_once). I'd go with choosing better where the logs should be placed and avoid redundancy.